nifi-nar-bundles/nifi-smb-bundle/nifi-smb-processors/src/main/java/org/apache/nifi/processors/smb/ListSmb.java

```diff
@@ -0,0 +1,353 @@
+            .displayName("Minimum File Age in milliseconds")
+            .name("min-file-age")
+            .description(
+                    "Any file younger then the given value will be omitted. Ideally this value should be greater then"
+                            + "the amount of time needed to perform a list.")
+            .required(true)
+            .addValidator(TIME_PERIOD_VALIDATOR)
+            .defaultValue("5ms")
+            .build();
+
+    public static final PropertyDescriptor MINIMUM_SIZE = new PropertyDescriptor.Builder()
+            .displayName("Minimum File Size in bytes")
+            .name("min-file-size")
+            .description("Any file smaller then the given value will be omitted.")
+            .required(false)
+            .addValidator(NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor MAXIMUM_SIZE = new PropertyDescriptor.Builder()
+            .displayName("Maximum File Size in bytes")
+            .name("max-file-size")
+            .description("Any file bigger then the given value will be omitted.")
+            .required(false)
+            .addValidator(NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+
+    public static final PropertyDescriptor SMB_LISTING_STRATEGY = new PropertyDescriptor.Builder()
+            .fromPropertyDescriptor(LISTING_STRATEGY)
+            .allowableValues(BY_ENTITIES, NO_TRACKING, BY_TIMESTAMPS)
+            .build();
+
+    public static final PropertyDescriptor SMB_CONNECTION_POOL_SERVICE = new Builder()
+            .name("smb-connection-pool-service")
+            .displayName("SMB Connection Pool Service")
+            .description("Specifies the SMB Connection Pool to use for creating SMB connections.")
+            .required(true)
+            .identifiesControllerService(SmbSessionProviderService.class)
+            .build();
+
+    public static final PropertyDescriptor SKIP_FILES_WITH_SUFFIX = new Builder()
+            .name("file-name-suffix-filter")
+            .displayName("File Name Suffix Filter")
+            .description("Files ends with the given suffix will be omitted. This is handy when writing large data into "
+                    + "temporary files and then moved to a final one. Please be advised that writing data into files "
+                    + "first is highly recommended when using Entity Tracking or Timestamp based listing strategies.")
+            .required(false)
+            .addValidator(NON_EMPTY_VALIDATOR)
+            .addValidator(new MustNotContainDirectorySeparatorsValidator())
+            .build();
+    public static final PropertyDescriptor SHARE = new PropertyDescriptor.Builder()
+            .displayName("Share")
+            .name("share")
+            .description("The network share to which files should be listed from. This is the \"first folder\"" +
+                    "after the hostname: smb://hostname:port\\[share]\\dir1\\dir2")
+            .required(false)
```

Comment:
It seems like `Share` should be a required property. Is it possible to connect without a Share?
```suggestion
            .required(true)
```